apphub-lib/apphub-lib-concurrency/src/main/java/com/github/saphyra/apphub/lib/concurrency/ExecutorServiceBean.java

```diff
@@ -68,41 +65,32 @@ public <I, R> List<R> processCollectionWithWait(List<I> dataList, Function<I, R>
         }
 
         return Lists.partition(dataList, parallelism)
-            .stream()
-            .map(part -> processCollectionWithWait(part, mapper))
-            .flatMap(Collection::stream)
-            .collect(Collectors.toList());
+                .stream()
+                .map(part -> processCollectionWithWait(part, mapper))
+                .flatMap(Collection::stream)
+                .collect(Collectors.toList());
     }
 
     public <I, R> List<R> processCollectionWithWait(Collection<I> dataList, Function<I, R> mapper) {
         log.debug("Processing {} items...", dataList.size());
 
         List<Future<ExecutionResult<R>>> futures = dataList.stream()
-            .map(i -> executor.submit(wrap(() -> mapper.apply(i))))
-            .collect(Collectors.toList());
-
-        long inProgress;
-        do {
-            inProgress = futures.stream()
-                .filter(rFuture -> !rFuture.isDone())
-                .count();
+                .map(i -> executor.submit(wrap(() -> mapper.apply(i))))
+                .collect(Collectors.toList());
 
-            if (inProgress > 0) {
-                log.debug("Incomplete tasks: {} out of {}", inProgress, dataList.size());
-                sleepService.sleep(1);
+        List<R> results = new ArrayList<>();
+        for (Future<ExecutionResult<R>> future : futures) {
+            try {
+                results.add(future.get().getOrThrow());
+            } catch (InterruptedException e) {
+                Thread.currentThread().interrupt();
```

Comment:
We don't need explicit errorHandling here. Please use @SneakyThrows on method instead